### PR TITLE
Add built-in DEBUG logging

### PR DESCRIPTION
**Pull request by harlowja**

### CODE REVIEW

File: tenacity/_utils.py [hunk 1]
```diff
@@ -36,6 +41,13 @@ def capture(fut, tb):
         fut.set_exception(tb[1])
 
 
+def pick_first_not_none(*args, **kwargs):
```
Comment (jd) [inline tenacity/_utils.py]: pip install first

:-)

**Comment by jd** [comment]
TRACE does not exist, I don't think it's a good idea to invent it. Just go ahead with DEBUG.

**Comment by harlowja** [comment]
So I've been there, yes, TRACE does not exist, but if its DEBUG, the openstack folks will flip out at us (because lots of people run openstack at DEBUG level, mainly because there is no other useful way to figure out what's happening in the code otherwise).

File: tenacity/_utils.py (hunk 1 above)
Comment (harlowja) [inline tenacity/_utils.py]: Meh :-P

**Comment by jd** [comment]
I think it's good to have this at DEBUG level anyway as if you run your service at that level to debug it, this is interesting option.

The rest of the "OpenStack" problem, well, they'll have to fix it in their projects. ;)

**Comment by harlowja** [comment]
Okie dokie, I'll adjust it soon :-P

**Pull request closed without merging**